extensions/sd_api_pictures/script.py

```diff
@@ -97,19 +152,51 @@ def input_modifier(string):
 
     global params
 
-    if not params['mode'] == 1:  # if not in immersive/interactive mode, do nothing
-        return string
-
-    if triggers_are_in(string):  # if we're in it, check for trigger words
+    if params['mode'] == 1:  # if not in immersive/interactive mode, do nothing
+            return if_of_is_in(string)
+    if params['mode'] == 2:
+        params['characterfocus'] = False
         toggle_generation(True)
```

Comment:
You shouldn't toggle the generation here at all because the current state could be SUPPRESSED (user has pressed the 'Supress pic' button while in Picturebook/Adventure mode)